### PR TITLE
fixes for chen/ledoitwolf

### PR DESCRIPTION
**Pull request by tlienart**
Happy to discuss this before merging.

PS: I'd feel better if the optimal shrinkage parameters were explicitly tested! Maybe one way is to explicitly compute it by hand for an ill conditioned matrix, another would be to take an example from their implementation?

### CODE REVIEW

File: src/chen.jl [hunk 1]
```diff
@@ -5,19 +5,32 @@ using LinearAlgebra
...
+        new{T}(s)
+    end
+    function RaoBlackwellLedoitWolf(s::Symbol=:optimal)
+        @assert s ∈ [:optimal] "Shrinkage setting not supported"
+        new{Symbol}(s)
+    end
 end
 
-RaoBlackwellLedoitWolf() = RaoBlackwellLedoitWolf{Symbol}(:auto)
+function chenoptimalshrinkage(n, p, C)
```
Comment (mateuszbaran) [inline src/chen.jl]: I think this function should actually be called `raoblackwellledoitwolfshrinkage` (to make a point that it is different from Oracle Approximating Shrinkage).

**Comment by mateuszbaran** [comment]
Nice changes! Adding tests based on a reference implementation is actually important, I'll add some soon.

File: src/chen.jl (hunk 1 above)
Comment (tlienart) [reply src/chen.jl]: sounds good, see latest commit

**Comment by tlienart** [comment]
I've extended the changes to OAS + changed the function names to reflect your earlier comment.

For the numerical tests, I'll open another PR soon but it may be good to merge this one already which is more API-ish.

**Comment by tlienart** [comment]
now superseded by #13. Closing this one.

**Pull request closed without merging**